apps/ledger-live-mobile/src/components/RequiresBLE/hooks/useAndroidBluetoothPermissions.ts

```diff
@@ -0,0 +1,191 @@
+};
+
+/**
+ * Requests the required permissions to use Bluetooth.
+ *
+ * On Android 11 and above, the app needs to have the BLUETOOTH_SCAN and BLUETOOTH_CONNECT permissions.
+ * On iOS, no request is done and returns true.
+ */
+async function requestBluetoothPermissions(): Promise<RequestMultipleResult> {
+  if (bluetoothPermissions.length === 0)
+    return { allGranted: true, generalStatus: RESULTS.GRANTED };
+
+  return PermissionsAndroid.requestMultiple(bluetoothPermissions).then(res => {
+    const permissionStatuses = Object.values(res);
+    const allGranted = permissionStatuses.every(
+      status => status === RESULTS.GRANTED,
+    );
+
+    /** https://developer.android.com/about/versions/11/privacy/permissions#dialog-visibility */
+    const someNeverAskAgain = permissionStatuses.some(
+      status => status === RESULTS.NEVER_ASK_AGAIN,
+    );
+
+    return {
+      allGranted,
+      generalStatus: allGranted
+        ? RESULTS.GRANTED
+        : someNeverAskAgain
+        ? RESULTS.NEVER_ASK_AGAIN
+        : RESULTS.DENIED,
+    };
+  });
+}
+
+/**
+ * Hook to check and request the BLE permission on Android.
+ *
+ * @returns an object with the following properties:
+ * - hasPermission: true if the permission is granted, false if it's denied, undefined if it's still loading
+ * - checkAgain: a function to check (and not request) again the permission
+ * - neverAskAgain: true if the user has checked the "never ask again" checkbox (or on Android 11+ if the user has denied the permission several times)
+ * - renderChildren: a boolean to know if the children should be rendered (permissions are granted) or not
+ */
+export const useAndroidBluetoothPermissions = () => {
+  const [hasPermission, setHasPermission] = useState<boolean | undefined>();
+  const [check, setCheck] = useState(false);
+  const [neverAskAgain, setNeverAskAgain] = useState(false);
+  const [retryCheckNonce, setRetryCheckNonce] = useState(0);
+  const [retryRequestNonce, setRetryRequestNonce] = useState(0);
+  const timeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  // Exposes a retry mechanism only of the permission check
+  const checkAgain = useCallback(() => {
+    setRetryCheckNonce(new Date().getTime());
+  }, []);
+
+  // Exposes a retry mechanism only of the permission request
+  const requestAgain = useCallback(() => {
+    setRetryRequestNonce(i => i + 1);
+    // setRetryRequestNonce(new Date().getTime());
+  }, []);
+
+  useEffect(() => {
+    // set hasPermission to false after 400ms if it's still undefined.
+    // without this we risk a black screen when this unmounts or a blink
+    // if we ignore it. Feel free to come up with a better way.
+    timeout.current = setTimeout(() => {
+      setHasPermission(false);
+    }, 400);
+
+    return () => {
+      if (timeout.current) clearTimeout(timeout.current);
+    };
+  }, []);
+
+  /**
+   * Goes to state="background" when for ex:
```

Comment:
Maybe document this in the body of the listener itself, and rephrase to something like "The value of the state will be "background" when for ex ...".
"Goes" is a bit ambiguous